File: backend/src/main/java/com/smartmoneymanager/backend/service/NotificationService.java
package com.smartmoneymanager.backend.service;

import java.util.List;

import com.smartmoneymanager.backend.dto.response.NotificationResponse;
import com.smartmoneymanager.backend.entity.enums.NotificationType;

public interface NotificationService {

    List<NotificationResponse> getNotifications(Long userId, boolean unreadOnly);

    long getUnreadCount(Long userId);

    NotificationResponse markAsRead(Long userId, Long notificationId);

    void markAllAsRead(Long userId);

    void deleteNotification(Long userId, Long notificationId);

    /** Internal API other services use to raise a notification for a user; not exposed over REST. */
    void notify(Long userId, NotificationType type, String title, String message);
}

File: frontend/src/services/notificationService.js
import apiClient from './apiClient.js';

export async function getNotifications(unreadOnly = false) {
  const { data } = await apiClient.get('/notifications', { params: { unreadOnly } });
  return data.data;
}

export async function getUnreadCount() {
  const { data } = await apiClient.get('/notifications/unread-count');
  return data.data.count;
}

export async function markAsRead(id) {
  const { data } = await apiClient.put(`/notifications/${id}/read`);
  return data.data;
}

export async function markAllAsRead() {
  await apiClient.put('/notifications/read-all');
}

export async function deleteNotification(id) {
  await apiClient.delete(`/notifications/${id}`);
}
